lib/routes/yamibo/bbs/forum.ts

```diff
@@ -0,0 +1,111 @@
+        type: '板块子分类，网页中选中板块分类后URL中的typeid值',
+    },
+    maintainers: ['KarasuShin'],
+    handler,
+    features: {
+        antiCrawler: true,
+        requireConfig: [
+            {
+                optional: true,
+                name: 'YAMIBO_SALT',
+                description:
+                    '百合会BBS登录后的认证信息，获取方式：1. 登录百合会BBS网页版 2. 打开浏览器开发者工具，切换到 Application 面板\n3. 点击侧边栏中的Storage -> Cookies -> https://bbs.yamibo.com 4. 复制 Cookie 中的 EeqY_2132_saltkey 值',
+            },
+            {
+                optional: true,
+                name: 'YAMIBO_AUTH',
+                description:
+                    '百合会BBS登录后的认证信息，获取方式：1. 登录百合会BBS网页版 2. 打开浏览器开发者工具，切换到 Application 面板\n3. 点击侧边栏中的Storage -> Cookies -> https://bbs.yamibo.com 4. 复制 Cookie 中的 EeqY_2132_auth 值',
+            },
+        ],
+    },
+    description: `:::warning
+百合会BBS访问部分板块需要用户登录认证，请参考配置说明
+:::`,
+};
+
+const origin = 'https://bbs.yamibo.com';
+
+async function handler(ctx: Context): Promise<Data> {
+    const fid = ctx.req.param('fid');
+    const type = ctx.req.param('type');
+    const { auth, salt } = config.yamibo;
+
+    const params = new URLSearchParams();
+    params.set('mod', 'forumdisplay');
+    params.set('fid', fid);
+    params.set('orderby', 'dateline');
+    if (type) {
+        params.set('filter', 'typeid');
+        params.set('typeid', type);
+    }
+    const headers: HeadersInit = {};
+
+    if (auth && salt) {
+        headers.cookie = `EeqY_2132_saltkey=${salt}; EeqY_2132_auth=${auth}`;
+    }
+
+    const link = `${origin}/forum.php?${params.toString()}`;
+
+    const $ = load(await ofetch<string>(link, { headers }));
+
+    const title = $('title').text().replace(' -  百合会 -  Powered by Discuz!', '');
+
+    let items: DataItem[] = $('tbody[id^="normalthread_"]')
+        .toArray()
+        .map((item) => {
+            const $item = $(item);
+            const id = $item.attr('id')!.match(/\d+/)![0];
+            const title = $item.find('th em').text() + $item.find('th .s.xst').text();
+            const link = `${origin}/thread-${id}-1-1.html`;
+            const pubDate = getDate($item.find('td.by').first().find('em').text());
+
+            return {
+                id,
+                title,
+                link,
+                pubDate,
+            };
+        });
+
+    items = await asyncPoolAll(10, items, async (item) => {
+        const description = (await cache.tryGet(item.link!, async () => {
+            const { data } = await fetchThread(item.id!);
+            if (data && !data.startsWith('<script type="text/javascript">')) {
+                const $ = load(data);
+                const op = $('#postlist>div[id^="post_"]').first();
+                const postId = op.attr('id')!.match(/\d+/)![0];
+                return op.find('table').find('tr').first().find(`#postmessage_${postId}`).parent().html()!;
+            }
+            return '';
+        })) as string;
```

Comment:
Tested against `/yamibo/bbs/forum/5/404`:
- All images are shown as placeholder image `none.gif`.
- Blank screen (first and third item, thread = 545850, 545442)
- Missing attachment (second item, thread = 545658)